liberum_core/src/lib/lib.rs

```diff
@@ -81,7 +84,7 @@ pub enum DaemonResponse {
     NodeList(Vec<NodeInfo>),
     FileProvided { id: String },
     Providers { ids: Vec<String> },
-    FileDownloaded { data: Vec<u8> }, // TODO ideally the data should not be a Vec<u8> but some kind of a stream to save it to disk instead of downloading the whole file in memory
+    FileDownloaded { data: PlainFileObject }, // TODO ideally the data should not be a Vec<u8> but some kind of a stream to save it to disk instead of downloading the whole file in memory
```

Comment:
The data is still a Vec<u8>, just hidden inside an object. Maybe this could be implemented in the object itself, but removing this TODO wouldn't be right